addons/website_sale/models/sale_order.py

```diff
@@ -243,6 +243,13 @@ def _update_address(self, partner_id, fnames=None):
             # Recompute taxes on fpos change
             self._recompute_taxes()
 
+        # If the user has explicitly selected a valid pricelist, we don't want to change it
+        if request.session.get('website_sale_selected_pl'):
+            selected_pricelist = request.session['website_sale_selected_pl']
+            valid_pl = self._valid_pricelist(selected_pricelist)
+            if valid_pl:
```

Comment:
```suggestion
            selected_pricelist = self.env['product.pricelist'].browse(selected_pricelist_id).exists()
            if (
                selected_pricelist
                and selected_pricelist._is_available_on_website(self.website_id)
                and selected_pricelist._is_available_in_country(self.partner_id.country_id.code)
            ):
```
Let's avoid adding yet another method to check the validity of a pricelist, please rely on the existing methods :pray:

Reply:
My bad! I'll make sure of it from now onwards :sweat_smile:

Reply:
Don't take it too strictly, it really depends from one situation to another, it can be cleaner and easier to read to have a specific method, but the existing logic is already complex enough here (and there are similar methods already).

Reply:
Yess! Sure :smiley: